doc/guides/timers-in-node.md

```diff
@@ -0,0 +1,175 @@
+with the system, and despite the fact that the API mirrors the browser 
+API, there are some differences in implementation.
+
+### "When I say so" Execution ~ *`setTimeout()`*
+
+`setTimeout()` can be used to schedule code execution after a designated
+amount of milliseconds. This function is also a part of the browser 
+JavaScript API and can be used in the exact same manner.
+
+`setTimeout()` accepts the code to execute as its first argument and the
+millisecond delay defined as a number literal as the second argument. Additional
+arguments may also be included and these will be passed on to the function. Here
+is an example of that:
+
+```js
+function myFunc (arg) {
+  console.log('arg was => ' + arg);
+}
+
+setTimeout(myFunc, 1500, 'funky');
+```
+
+The above function `myFunc()` will execute as close to 1500
+milliseconds (or 1.5 seconds) as possible due to the call of `setTimeout()`.
+
+The timeout interval that is set cannot be relied upon to execute after
+that *exact* number of milliseconds. This is because other executing code that
+blocks or holds onto the event loop will push the execution of the timeout
+back. The *only* guarantee is that the timeout will not execute *sooner* than
+the declared timeout interval.
+
+`setTimeout()` returns a `Timeout` object that can be used to reference the
+timeout that was set. This returned object can be used to cancel the timeout (
+see `clearTimeout()` below) as well as change the execution behavior (see
+`unref()` below).
+
+### "Right after this" Execution ~ *`setImmediate()`*
+
+`setImmediate()` will execute code at the beginning of the next
+event loop cycle. This code will execute *before* any timers or IO operations.
+This code execution could be thought of as happening "right after this", meaning
+any code following the `setImmediate()` function call will execute before the
+`setImmediate()` function argument. Here's an example:
+
+```js
+console.log('before immediate');
+
+setImmediate(() => {
+  console.log('executing immediate');
+});
+
+console.log('after immediate');
+```
+
+The above function passed to `setImmediate()` will execute after all runnable
+code has executed, and the console output will be:
+
+```shell
+before immediate
+after immediate
+executing immediate
+```
+
+`setImmediate()` returns and `Immediate` object, which can be used to cancel
+the scheduled immediate (see `clearImmediate()` below).
+
+Note: Don't get `setImmediate()` confused with `process.nextTick()`. There are
+some major ways they differ. The first is that `process.nextTick()` will run
+*before* any `Immediate`s that are set as well as before any scheduled IO.
+The second is that `process.nextTick()` is non-clearable, meaning once
+code has been scheduled to execute with `process.nextTick()`, the execution
+cannot be stopped, just like with a normal function. In fact the code passed to
+`process.nextTick()` will be run asynchronously, and thus outside the current
+call stack.
```

Comment:
`process.nextTick()` is like calling the function synchronously, only with out the current call stack context.